### DISCUSSION TITLE
m_wrappingDelta and Initial SQN values

### DISCUSSION THREAD

**Discussion by laurent-dubesset** [aligungr/UERANSIM#600] [Q&A] [answered]
Hello,

m_wrappingDelta value: what is the values please?
Initial SQN value : where can I find the initial value of SQN when I run UERANSIM for the first time and or a given IMSI?
If I stop the software and then run it again, do you store the SQN somewhere to avoid being out of SYNCH with the UDM stored value or will you simply let the SYNCHONIZATION procedure run?

Thanks

**Comment by abousselmi** [comment, marked as answer]
Hi Laurent,

Regarding the SQN wrapping delta, SqnManager takes two values as input
https://github.com/aligungr/UERANSIM/blob/1d184a73749634d02f9bc2285fcc0508ff0cbbd9/src/ue/nas/usim/sqn_mng.hpp#L29

The second one is the wrappingDelta, which in terns can be found here: https://github.com/aligungr/UERANSIM/blob/1d184a73749634d02f9bc2285fcc0508ff0cbbd9/src/ue/nas/usim/usim.cpp#L18

`1ull << 28ull = 268435456`

Cheers,
